airgun/entities/repository.py

```diff
@@ -61,9 +63,34 @@ def update(self, product_name, entity_name, values):
         if values.get('repo_content.http_proxy_policy') == 'Global Default':
             values['repo_content.http_proxy_policy'] = self.global_default_http_proxy
         view = self.navigate_to(self, 'Edit', product_name=product_name, entity_name=entity_name)
+        # Dismiss any existing flash messages before filling to avoid confusion
+        with suppress(NoSuchElementException, AttributeError):
+            view.flash.dismiss()
         view.fill(values)
-        view.flash.assert_no_error()
-        view.flash.dismiss()
+        # Wait for all EditableEntry auto-saves to complete and flash messages to stabilize
+        # EditableEntry widgets auto-save individually, so we need to wait for UI to settle
+        # before checking flash messages to avoid reading non-existent toast messages
+        wait_for(
+            lambda: view.browser.plugin.ensure_page_safe(timeout='1s') is None,
+            handle_exception=True,
+            timeout=30,
+            delay=1,
+            logger=view.flash.logger,
+        )
```

Comment:
```suggestion
self.browser.plugin.ensure_page_safe(timeout='30s')
```
Can simply do this, should make it quite a bit more readable.